rmw_zenoh_cpp/src/detail/rmw_service_data.cpp

```diff
@@ -447,21 +435,33 @@ rmw_ret_t ServiceData::send_response(
 
   size_t data_length = ser.get_serialized_data_length();
 
-  const z_loaned_query_t * loaned_query = query->get_query();
-  z_query_reply_options_t options;
-  z_query_reply_options_default(&options);
-
-  z_owned_bytes_t attachment;
-  rmw_zenoh_cpp::create_map_and_set_sequence_num(
-    &attachment, request_id->sequence_number,
-    request_id->writer_guid);
-  options.attachment = z_move(attachment);
+  const zenoh::Query & loaned_query = query->get_query();
+  zenoh::Query::ReplyOptions options = zenoh::Query::ReplyOptions::create_default();
+  std::vector<uint8_t> writer_gid;
+  for (size_t i = 0; i < RMW_GID_STORAGE_SIZE; ++i) {
+    writer_gid.push_back(request_id->writer_guid[RMW_GID_STORAGE_SIZE - 1 - i]);
+  }
+  options.attachment = create_map_and_set_sequence_num(
+    request_id->sequence_number,
+    writer_gid);
+
+  std::vector<uint8_t> raw_bytes(
+    reinterpret_cast<const uint8_t *>(response_bytes),
+    reinterpret_cast<const uint8_t *>(response_bytes) + data_length);
+  zenoh::Bytes payload(raw_bytes);
```

Comment:
Actually we should use `zenoh::Bytes payload(std::move(raw_bytes));` to avoid a copy. See [here](https://github.com/eclipse-zenoh/zenoh-cpp/blob/94edfd9845b63bab4761be328bd64db95daab212/include/zenoh/api/bytes.hxx?plain=1#L65) for details.